### PR TITLE
[multimod] Resolve version for commit per module instead of per module set

### PR DESCRIPTION
**Pull request by mx-psi**
Fixes open-telemetry/opentelemetry-collector/issues/9827

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

File: multimod/internal/common/tools.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ func GetModuleSet(modSetName, versioningFilename string) (ModuleSet, error) {
 
 // updateGoModVersions updates one go.mod file, given by modFilePath, by updating all modules listed in
 // newModPaths to use the newVersion given.
```
Comment (pellared) [inline multimod/internal/common/tools.go:63]: Comment needs to be updated to reflect the change.

**Review by pellared — APPROVED** [review]

**Comment by mx-psi** [comment]
I tried this out on open-telemetry/opentelemetry-collector-contrib/pull/33977. It works correctly, although it is a bit slower at first and faster at the tidying stage (likely because the Go proxy is caching the version at an earlier stage). The overall time feels about the same.

File: multimod/internal/common/tools.go (hunk 1 above)
Comment (mx-psi) [reply multimod/internal/common/tools.go]: Fixed, thanks!

**Review by codeboten — APPROVED** [review]
Thanks @mx-psi!

**Pull request merged**